blockly/src/server/Server.java

```diff
@@ -1287,4 +1372,47 @@ public void interact() {
                         interactionComponent ->
                             interactionComponent.triggerInteraction(entity, hero)));
   }
+
+  /**
+   * Attempts to push an entity in front of the hero.
+   *
+   * <p>If there is a pushable entity in the tile in front of the hero, it checks if the tile behind
+   * it is accessible. If accessible, the hero and the entity are moved simultaneously in the
+   * direction the hero is facing.
+   *
+   * <p>The pushed entity temporarily loses its blocking component while moving and regains it
+   * after.
+   */
+  public void push() {
+    PositionComponent pc =
+        hero.fetch(PositionComponent.class)
+            .orElseThrow(() -> MissingComponentException.build(hero, PositionComponent.class));
+    Tile inFront = Game.tileAT(pc.position(), pc.viewDirection());
+    // assumption only one pushable per tile
+    Optional<Entity> pushable =
+        Game.entityAtTile(inFront).filter(e -> e.isPresent(PushableComponent.class)).findFirst();
+    if (pushable.isPresent()) {
```

Comment:
hmm. ich finde die logik schwer lesbar: im schritt vorher filterst du nach der `PushableComponent` und schließt den stream ab mit `findCirst`. jetzt musst du das optional auspacken, den fetch für die `PositionComponent` machen und dann evtl eine exception werfen...

was passiert, wenn du den stream oben nicht abschließt, sondern mit einem `map` die fetch-operation ausführst und von diesem ergebnis ggf. das erste nimmst? spart ein paar anweisungen und macht es lesbarer? dann ist das ergebnis direkt die gesuchte `PositionComponent`?

Reply:
hmmm. den rest kriegt man vermutlich auch noch in den selben stream rein :)

Reply:
Kann ich gucken. Gefühlt geht mit streams ja alles irgendwie

Reply:
versteh mich nicht falsch, mir eght es nicht um